appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockBall.java

```diff
@@ -99,21 +106,26 @@ private void setZProperty(String text) {
       mockCanvas.reorderComponents(this);
     }
   }
-  
-  private void setXProperty(String text) {
+
+  private void refreshCanvas() {
     MockCanvas mockCanvas = (MockCanvas) getContainer();
     // mockCanvas will be null for the MockBall on the palette
     if (mockCanvas != null) {
-      mockCanvas.reorderComponents(this);
+      mockCanvas.refreshForm();
```

Comment:
My change was meant to be an incremental improvement over the call to `reorderComponents()`, which, as you may know, is:

```  void reorderComponents(MockSprite changedSprite) {
    // Since it has already been added to rootPanel, we do not need to pass it,
    // just to re-sort the list of rootPanel's children.
    sortSprites();

    // Redraw.
    refreshForm();
  }
```

While further improvement may be possible, I am less sure of that change and would rather not have it in this PR. (I've already put more into it than ideal.)

Reply:
Ok. That makes sense to me. Thanks for clarifying.